web/src/components/tasks/LicenseTask.test.tsx

```diff
@@ -55,417 +60,678 @@ describe("<LicenseTask />", () => {
-    const returnedPromise = Promise.resolve(generateUserData({}));
-    mockApi.checkLicenseStatus.mockReturnValue(returnedPromise);
-    renderTask();
-
-    fillText("business-name", "My Cool Nail Salon");
-    fillText("address-1", "123 Main St");
-    fillText("zipcode", "12345");
-
-    expect(screen.queryByTestId("loading-spinner")).not.toBeInTheDocument();
-
-    fireEvent.submit(screen.getByTestId("check-status-submit"));
-    expect(screen.getByTestId("loading-spinner")).toBeInTheDocument();
-
-    await waitFor(() => {
-      expect(screen.queryByTestId("loading-spinner")).not.toBeInTheDocument();
-    });
-  });
-
-  it("displays the loading spinner while failed request is being made", async () => {
-    const returnedPromise = Promise.reject(404);
-    mockApi.checkLicenseStatus.mockReturnValue(returnedPromise);
-    renderTask();
-
-    fillText("business-name", "My Cool Nail Salon");
-    fillText("address-1", "123 Main St");
-    fillText("zipcode", "12345");
-
-    expect(screen.queryByTestId("loading-spinner")).not.toBeInTheDocument();
-
-    fireEvent.submit(screen.getByTestId("check-status-submit"));
-    expect(screen.getByTestId("loading-spinner")).toBeInTheDocument();
-    await act(() => {
-      return returnedPromise.catch(() => {});
+      it("displays FIELDS_REQUIRED error alert when some information is missing", async () => {
+        useMockBusiness({
+          licenseData: generateLicenseData(
+            {},
+            {
+              [taskIdLicenseNameMapping[task.id]]: generateLicenseDetails({
+                lastUpdatedISO: "",
+                nameAndAddress: createEmptyLicenseSearchNameAndAddress(),
+              }),
+            }
+          ),
+        });
+        renderTask();
+        fireEvent.click(screen.getByText(Config.licenseSearchTask.tab2Text));
+        expect(screen.queryByTestId("error-alert-FIELDS_REQUIRED")).not.toBeInTheDocument();
+        fireEvent.submit(screen.getByTestId("check-status-submit"));
+        await waitFor(() => {
+          expect(screen.getByTestId("error-alert-FIELDS_REQUIRED")).toBeInTheDocument();
+        });
+        expect(mockApi.checkLicenseStatus).not.toHaveBeenCalled();
+      });
     });
-    expect(screen.queryByTestId("loading-spinner")).not.toBeInTheDocument();
-  });
-
-  describe("receipt screen", () => {
-    it("displays license status results when it is found", async () => {
-      renderTask();
 
-      mockApi.checkLicenseStatus.mockResolvedValue(
-        generateUserDataForBusiness(
-          generateBusiness({
-            licenseData: generateLicenseData({
-              items: [
-                generateLicenseStatusItem({ title: "application fee", status: "PENDING" }),
-                generateLicenseStatusItem({ title: "board approval", status: "ACTIVE" }),
-              ],
-            }),
-          })
-        )
-      );
+    describe("loading spinner", () => {
```

Comment:
Is there a better way to right this test.. I've added the SetTimeout due to flakey failing test. Even with the 1000ms delay and { timeout: 5000 } wait time added to the assertion the tests are still flakey